modules/extensions-support/src/test/java/org/mule/module/extension/internal/AnnotationsBasedDescriberTestCase.java

```diff
@@ -361,4 +377,57 @@ public void setExtendedProperty(String extendedProperty)
             this.extendedProperty = extendedProperty;
         }
     }
+
+    @org.mule.extension.annotations.Extension(name = EXTENSION_NAME, description = EXTENSION_DESCRIPTION, version = EXTENSION_VERSION)
+    @Extensible
+    public static class ExtensibleExtension
+    {
+
+    }
+
+    @org.mule.extension.annotations.Extension(name = EXTENSION_NAME, description = EXTENSION_DESCRIPTION, version = EXTENSION_VERSION)
+    @Operations({ClassLevelExtensionOfOperation.class, MethodLevelExtensionOfOperation.class})
+    public static class ExtendingExtension
+    {
+
+    }
+
+    @org.mule.extension.annotations.Extension(name = EXTENSION_NAME, description = EXTENSION_DESCRIPTION, version = EXTENSION_VERSION)
+    @Operations({MethodLevelExtensionOfOperation.class, ExtensibleExtensionOperation.class})
+    @Extensible
+    public static class ExtensibleExtendingExtension
+    {
+
+    }
+
+    @ExtensionOf(ExtensibleExtension.class)
+    private static class ClassLevelExtensionOfOperation
+    {
+        @Operation
+        public String sayHello()
+        {
+            return "Hello!";
+        }
+    }
+
+
+    private static class MethodLevelExtensionOfOperation
+    {
+        @ExtensionOf(ExtensibleExtension.class)
+        @Operation
+        public String sayBye()
+        {
+            return "Bye!";
+        }
+    }
+
+    private static class ExtensibleExtensionOperation
```

Comment:
this class got way too big... Can you split it out to another test which specifically focuses on extensible/extensionOf concerns?